__tests__/app.test.js

```diff
@@ -158,6 +158,90 @@ describe('GET /api/articles', () =>
+    {
+        const userCommentReference = 
+        {
+            username: 'icellusedkars',
+            body: 'new comment test'
+        }
+        return request(app).post('/api/articles/1/comments').expect(201).send(userCommentReference)
+        .then(({ body }) =>
+        {
+            expect(Array.isArray(body)).toBe(false)
+            expect(typeof body).toBe('object')
+        })
+    })
+    test(`the returned object contains the properties comment_id(number), body(string), article_id(number),
+    author(string), votes(number), created_at(string)`, () =>
+    {
+        const userCommentReference = 
+        {
+            username: 'icellusedkars',
+            body: 'new comment test'
+        }
+        return request(app).post('/api/articles/1/comments').expect(201).send(userCommentReference)
+        .then(({ body }) =>
+        {
+            expect(body).toEqual
+            (
+                expect.objectContaining(
+                    {
+                        comment_id: expect.any(Number),
+                        body: expect.any(String),
+                        article_id: expect.any(Number),
+                        author: expect.any(String),
+                        votes: expect.any(Number),
+                        created_at: expect.any(String)
+                    }
+                )
+            )
+        })
+    })
+    test(`returns the correct body, article_id and author`, () =>
+    {
+        const userCommentReference = 
+        {
+            username: 'icellusedkars',
+            body: 'new comment test'
+        }
+        return request(app).post('/api/articles/1/comments').expect(201).send(userCommentReference)
+        .then(({ body }) =>
+        {
+            expect(body.author).toEqual(userCommentReference.username)
+            expect(body.body).toEqual(userCommentReference.body)
+            expect(body.article_id).toEqual(1)
+        })
+    })
+    test('if article id given is not a number, return status 400 and error message', () =>
+    {
+        const userCommentReference = 
+        {
+            username: 'icellusedkars',
+            body: 'new comment test'
+        }
+        return request(app).post('/api/articles/test/comments').expect(400).send(userCommentReference)
+        .then((err) =>
+        {
+            expect(err.body.msg).toEqual('invalid article ID: not a number')
+        })
+    })
+    test('if trying to add a comment to an article id that does not exist then return status 404 and error message', () =>
+    {
+        const userCommentReference = 
+        {
+            username: 'icellusedkars',
+            body: 'new comment test'
+        }
+        return request(app).post('/api/articles/12098376/comments').expect(404).send(userCommentReference)
+        .then((err) =>
+        {
+            expect(err.body.msg).toEqual('invalid article ID: article not found')
+        })
+    })
+})
```

Comment:
this looks good Jace. Some extra tests needed: 

201 - ignores extra properties on request body
400 - missing body/username properties 
404 - username not found